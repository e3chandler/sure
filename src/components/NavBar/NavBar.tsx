import { Link, Box, Button } from '@mui/material';
import { Link as RouterLink, NavLink } from 'react-router-dom';
type TNavBar = {
  links: {
    text: string;
    href: string;
    'data-testid'?: string;
  }[];
};

function NavBar({ links }: TNavBar) {
  return (
    <Box
      component="aside"
      sx={{
        background: '#0c2975',
        padding: '16px',
        width: '200px',
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
      }}
    >
      <Link
        component={RouterLink}
        to="/"
        sx={{ cursor: 'pointer', marginBottom: '80px', marginTop: '40px' }}
      >
        <img src="/surelogo.svg" alt="logo"></img>
      </Link>

      {/* Use NavLink for the built in active class handling and accessibility */}
      {links.map(({ text, href, 'data-testid': dataTestId }) => (
        <Button
          component={NavLink}
          key={href}
          to={href}
          sx={{
            color: '#fff',  
            '&.active': {
              backgroundColor: '#0070ff'
            }
          }}
          data-testid={dataTestId}
        >
          {text}
        </Button>
      ))}
    </Box>
  );
}

export default NavBar;
