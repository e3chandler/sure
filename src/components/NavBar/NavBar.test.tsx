import NavBar from './NavBar';
import { renderWithProviders } from '../../utils/test';

describe('NavBar', () => {
  const defaultProps = {
    links: [
      { text: 'Link1', href: '/link1' },
      { text: 'Link2', href: '/link2' },
      { text: 'Link3', href: '/link3' },
    ],
  };

  it('should render NavBar links', () => {
    const { getByText } = renderWithProviders(<NavBar {...defaultProps} />);

    expect(getByText('Link1')).toBeInTheDocument();
    expect(getByText('Link2')).toBeInTheDocument();
    expect(getByText('Link3')).toBeInTheDocument();
  });

  // TODO: Challenge 2
  it('should render an `href` attribute for each link', () => {
    const { container } = renderWithProviders(<NavBar {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  // Challenge 2 again.
  it('should render an `href` attribute for each link - non snapshot', () => {
    const { getByText } = renderWithProviders(<NavBar {...defaultProps} />);
    const link1 = defaultProps.links[0]
    expect(getByText(link1.text).closest('a')).toHaveAttribute('href', link1.href)

    const link2= defaultProps.links[1]
    expect(getByText(link2.text).closest('a')).toHaveAttribute('href', link2.href)
    
    const link3 = defaultProps.links[2]
    expect(getByText(link3.text).closest('a')).toHaveAttribute('href', link3.href)
  })
});
