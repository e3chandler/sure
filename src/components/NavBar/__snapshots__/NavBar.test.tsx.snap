// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render an \`href\` attribute for each link 1`] = `
<div>
  <aside
    class="MuiBox-root css-1231o5f"
  >
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-x6o2tr-MuiTypography-root-MuiLink-root"
      href="/"
    >
      <img
        alt="logo"
        src="/surelogo.svg"
      />
    </a>
    <a
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-13286rk-MuiButtonBase-root-MuiButton-root"
      href="/link1"
      tabindex="0"
    >
      Link1
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
    <a
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-13286rk-MuiButtonBase-root-MuiButton-root"
      href="/link2"
      tabindex="0"
    >
      Link2
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
    <a
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-13286rk-MuiButtonBase-root-MuiButton-root"
      href="/link3"
      tabindex="0"
    >
      Link3
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
  </aside>
</div>
`;
